Jellyfin.Api/Controllers/NotificationsController.cs

```diff
@@ -0,0 +1,176 @@
+        [ProducesResponseType(typeof(IEnumerable<NotificationResultDto>), StatusCodes.Status200OK)]
+        public IActionResult GetNotifications(
+            [FromRoute] string userId,
+            [FromQuery] bool? isRead,
+            [FromQuery] int? startIndex,
+            [FromQuery] int? limit)
+        {
+            return Ok(new NotificationResultDto());
+        }
+
+        /// <summary>
+        /// Endpoint for getting a user's notification summary.
+        /// </summary>
+        /// <param name="userId">The user's ID.</param>
+        /// <returns>Notifications summary for the user.</returns>
+        [HttpGet("{UserID}/Summary")]
+        [ProducesResponseType(typeof(NotificationsSummaryDto), StatusCodes.Status200OK)]
+        public IActionResult GetNotificationsSummary(
+            [FromRoute] string userId)
+        {
+            return Ok(new NotificationsSummaryDto());
+        }
+
+        /// <summary>
+        /// Endpoint for getting notification types.
+        /// </summary>
+        /// <returns>All notification types.</returns>
+        [HttpGet("Types")]
+        [ProducesResponseType(typeof(IEnumerable<NameIdPair>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status500InternalServerError)]
+        public IActionResult GetNotificationTypes()
+        {
+            try
+            {
+                return Ok(_notificationManager.GetNotificationTypes());
+            }
+            catch (Exception e)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Endpoint for getting notification services.
+        /// </summary>
+        /// <returns>All notification services.</returns>
+        [HttpGet("Services")]
+        [ProducesResponseType(typeof(IEnumerable<NameIdPair>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status500InternalServerError)]
+        public IActionResult GetNotificationServices()
+        {
+            try
+            {
+                return Ok(_notificationManager.GetNotificationServices());
+            }
+            catch (Exception e)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Endpoint to send a notification to all admins.
+        /// </summary>
+        /// <param name="name">The name of the notification.</param>
+        /// <param name="description">The description of the notification.</param>
+        /// <param name="url">The URL of the notification.</param>
+        /// <param name="level">The level of the notification.</param>
+        /// <returns>Status.</returns>
+        [HttpPost("Admin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status500InternalServerError)]
```

Comment:
We don't need to document the possibility of 500 error responses in the API documentation